Game/MyProject/Game/Camera.cpp

```diff
@@ -45,35 +23,62 @@ IvMatrix44 Camera::getViewMatrix()
 IvMatrix44 Camera::getProjectionMatrix()
 {
     IvMatrix44 perspective;
-    
-    float d = 1.0f/IvTan(fieldOfView/180.0f*kPI*0.5f);
+    float aspectRatio = (float) screenWidth / screenHeight;
+    float d = 1.0f / IvTan(fieldOfView / 180.0f * kPI * 0.5f);
     
     perspective(0,0) = d / aspectRatio;
     perspective(1,1) = d ;
-    perspective(2,2) = (near+far) / (near-far);
-    perspective(2,3) = (2*near*far) / (near-far);
+    perspective(2,2) = (nearPlane + farPlane) / (nearPlane - farPlane);
+    perspective(2,3) = (2 * nearPlane * farPlane) / (nearPlane - farPlane);
     perspective(3,2) = -1.0f;
     perspective(3,3) = 0.0f;
     
     return perspective;
 }
 
+void Camera::zoom(float step)
+{
+    position += (lookAt - position) * step;
+}
 
-void Camera::setPosition(IvVector3 newPos)
+void Camera::pan(const IvVector2& offset)
+{
+    IvVector3 view = lookAt - position;
+    IvVector3 right =  view.Cross( orientation );
+    right.Normalize();
+    IvVector3 viewUp =  right.Cross( view );
+    viewUp.Normalize();
+    
+    position -= right * offset.x - viewUp * offset.y;
+    lookAt -= right * offset.x - viewUp * offset.y;
+}
+
+void Camera::rotate(const IvVector2& offset, IvVector3 point)
```

Comment:
you are working in 3D space. rotations should be either:
- degrees around an arbitrary axis
- euler angles (!careful susceptible to gimbal lock)
- quaternions (recommended)